bundles/binding/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/config/KNXGenericBindingProvider.java

```diff
@@ -380,10 +386,18 @@ protected KNXBindingConfig parseBindingConfigString(Item item, String bindingCon
                     throw new BindingConfigParseException("Only one readable GA allowed.");
                 }
 
-                Class<? extends Type> typeClass = item.getAcceptedCommandTypes().size() > 0
-                        ? item.getAcceptedCommandTypes().get(i)
-                        : item.getAcceptedDataTypes().size() > 1 ? item.getAcceptedDataTypes().get(i)
-                                : item.getAcceptedDataTypes().get(0);
+                Class<? extends Type> typeClass = null;
+                if( item.getAcceptedCommandTypes().size() > i ) {
+                  typeClass = item.getAcceptedCommandTypes().get(i);
+                }
+                else {
+                  if( item.getAcceptedDataTypes().size() > i ) {
+                    typeClass = item.getAcceptedDataTypes().get(i);
+                  }
+                  else {
+                    typeClass = item.getAcceptedDataTypes().get(0);
+                  }
+                }
```

Comment:
Is there a way to compatibly "relax" the current binding config string syntax so that the proper `Type` is chosen in cases where an `ArrayIndexOutOfBounds` would otherwise occur?  It's tricky, because this moves into the "do what I'm thinking" territory.